src/components/helpers/Audio.js

```diff
@@ -1,30 +1,38 @@
 import { Audio } from 'expo';
-import { forEach, map } from 'ramda';
+import { all, forEach, filter, identity, map, xprod } from 'ramda';
 import { Component } from 'react';
 
 export const createLoadSounds = Audio => {
   class LoadSounds extends Component {
     constructor(props) {
       super(props);
 
-      this.sounds = map(source => {
-        const sound = new Audio.Sound();
-        sound.loadAsync(source);
-        return sound;
-      }, this.props.sounds || []);
+      this.sources = filter(identity, this.props.sounds || []);
+      this.sounds = map(_source => new Audio.Sound(), this.sources);
+      this.state = { soundsLoaded: false };
+    }
+
+    componentDidMount() {
+      Promise.all(
+        map(
+          ([sound, source]) => sound.loadAsync(source),
+          xprod(this.sounds, this.sources)
```

Comment:
I think this should be `zip`, not `xprod`? Or even more directly judging by the documentation we could use `zipWith` like so:

```
Promise.all(
  zipWith((sound, source) => sound.loadAsync(source), this.sounds, this.sources)
)
).then(() => this.setState({ soundsLoaded: true }));
```

Reply:
😓 yep